client/asset/btc/btc.go

```diff
@@ -875,36 +881,46 @@ func newRPCWallet(requester RawRequesterWithContext, cfg *BTCCloneCFG, walletCon
 	return &ExchangeWalletFullNode{btc}, nil
 }
 
-func newUnconnectedWallet(cfg *BTCCloneCFG, walletCfg *WalletConfig) (*baseWallet, error) {
+func validateBtcWalletConf(walletCfg *WalletConfig) error {
 	// If set in the user config, the fallback fee will be in conventional units
 	// per kB, e.g. BTC/kB. Translate that to sats/byte.
-	fallbackFeesPerByte := toSatoshi(walletCfg.FallbackFeeRate / 1000)
-	if fallbackFeesPerByte == 0 {
-		fallbackFeesPerByte = cfg.DefaultFallbackFee
+	walletCfg.FallbackFeeRatePerByte = toSatoshi(walletCfg.FallbackFeeRate / 1000)
+	if walletCfg.FallbackFeeRatePerByte == 0 {
+		return fmt.Errorf("fallback fee rate limit is smaller than smallest unit(1e-5 coins/kB): %v",
+			walletCfg.FallbackFeeRate)
 	}
-	cfg.Logger.Tracef("Fallback fees set at %d %s/vbyte",
-		fallbackFeesPerByte, cfg.WalletInfo.UnitInfo.AtomicUnit)
-
 	// If set in the user config, the fee rate limit will be in units of BTC/KB.
 	// Convert to sats/byte & error if value is smaller than smallest unit.
-	feesLimitPerByte := cfg.DefaultFeeRateLimit
-	if feesLimitPerByte == 0 {
-		feesLimitPerByte = defaultFeeRateLimit
-	}
-	if walletCfg.FeeRateLimit > 0 {
-		feesLimitPerByte = toSatoshi(walletCfg.FeeRateLimit / 1000)
-		if feesLimitPerByte == 0 {
-			return nil, fmt.Errorf("fee rate limit is smaller than smallest unit: %v",
-				walletCfg.FeeRateLimit)
-		}
+	walletCfg.FeeRateLimitPerByte = toSatoshi(walletCfg.FeeRateLimit / 1000)
+	if walletCfg.FeeRateLimitPerByte == 0 {
+		return fmt.Errorf("fee rate limit is smaller than smallest unit(1e-5 coins/kB): %v",
+			walletCfg.FeeRateLimit)
+	}
+
+	if walletCfg.RedeemConfTarget == 0 {
+		return fmt.Errorf("redeem conf target can not be 0: %v",
+			walletCfg.RedeemConfTarget)
 	}
-	cfg.Logger.Tracef("Fees rate limit set at %d sats/byte", feesLimitPerByte)
 
-	redeemConfTarget := walletCfg.RedeemConfTarget
-	if redeemConfTarget == 0 {
-		redeemConfTarget = defaultRedeemConfTarget
+	return nil
+}
+
+func newUnconnectedWallet(cfg *BTCCloneCFG, walletCfg *WalletConfig) (*baseWallet, error) {
+	// if values not specified, use defaults. As they are validated as BTC/KB,
+	// we need to convert first.
+	if walletCfg.FallbackFeeRate == 0 {
+		walletCfg.FallbackFeeRate = float64(cfg.DefaultFallbackFee) * 1000 / 1e8
+	}
+	if walletCfg.FeeRateLimit == 0 {
+		walletCfg.FeeRateLimit = float64(cfg.DefaultFeeRateLimit) * 1000 / 1e8
+	}
+	if walletCfg.RedeemConfTarget == 0 {
+		walletCfg.RedeemConfTarget = defaultRedeemConfTarget
+	}
```

Comment:
If you're going to assign defaults here, you'll need want to do this in `Create` also, no? Maybe just move it in to `validateBtcWalletConf`.